.github/ISSUE_TEMPLATE/new_lint.md

```diff
@@ -0,0 +1,43 @@
+---
+name: New lint suggestion
+about: Suggest a new Clippy lint.
+labels: L-lint
+---
+
+### What it does
+
+*What does this lint do?*
+
+### Categories
```

Comment:
```suggestion
### Categories (optional)
```

I wouldn't force people to choose a group, since this is normally best done by experienced Clippy devs, instead of everyone who has a lint idea.